src/gobstuf/api.py

```diff
@@ -105,28 +124,31 @@ def _stuf():
     audit_logger = AuditLogger.get_instance()
 
     request = flask.request
-
-    method = request.method
-    assert method in ['GET', 'POST'], f"Unknown method {method}, GET or POST required"
-
     url = _routed_url(request.url)
-    if method == 'GET':
-        response = _get_stuf(url)
-    elif method == 'POST':
-        data = _update_request(request.data.decode())
-        response = _post_stuf(url, data, request.headers)
 
-    text = response.text
-    text = _update_response(text)
+    request_uuid = str(uuid.uuid4())
 
-    extra_data = {
+    request_log_data = {
         'soapaction': request.headers.get('Soapaction'),
-        'remote_response_code': response.status_code,
         'original_url': request.url,
-        'method': method,
+        'method': request.method,
     }
-
-    audit_logger.log_request(request.remote_addr, url, extra_data)
+    audit_logger.log_request(request.remote_addr, url, request_log_data, request_uuid)
+
+    response_log_data = {**request_log_data}
+
+    try:
+        response = _handle_stuf_request(request, url)
+    except Exception as e:
```

Comment:
Exception is too wide, I would propose except BadRequest and MethodNotAllowed so that we are sure that no sensitive data is ever posted to the audittrail